src/main/java/org/dependencytrack/tasks/OSVDownloadTask.java

```diff
@@ -0,0 +1,277 @@
+            this.isEnabled = enabled != null && Boolean.valueOf(enabled.getPropertyValue());
+        }
+    }
+
+    @Override
+    public void inform(Event e) {
+
+        if (e instanceof GoogleOSVMirrorEvent && this.isEnabled) {
+
+            for (Ecosystem ecosystem : Ecosystem.values()) {
+                LOGGER.info("Updating datasource with Google OSV advisories for ecosystem " + ecosystem.getValue());
+                String url = "https://osv-vulnerabilities.storage.googleapis.com/" + ecosystem.getValue() + "/all.zip";
+                request = new HttpGet(url);
+                request.setHeader("Accept", "*/*");
+                request.setHeader("User-Agent", "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.11 (KHTML, like Gecko) Chrome/23.0.1271.95 Safari/537.11");
+                try (final CloseableHttpResponse response = HttpClientPool.getClient().execute(request)) {
+                    final StatusLine status = response.getStatusLine();
+                    if (status.getStatusCode() == 200) {
+                        try (InputStream in = response.getEntity().getContent();
+                            ZipInputStream zipInput = new ZipInputStream(in)) {
+                            unzipFolder(zipInput);
+                        }
+                    } else {
+                        LOGGER.error("Download failed : " + status.getStatusCode() + ": " + status.getReasonPhrase());
+                    }
+                } catch (Exception ex) {
+                    LOGGER.error("Exception while executing Http client request", ex);
+                }
+            }
+        }
+    }
+
+    private void unzipFolder(ZipInputStream zipIn) throws IOException {
+
+        BufferedReader reader = new BufferedReader(new InputStreamReader(zipIn));
+        GoogleOSVAdvisoryParser parser = new GoogleOSVAdvisoryParser();
+        ZipEntry zipEntry = zipIn.getNextEntry();
+        while (zipEntry != null) {
+
+            String line = null;
+            StringBuilder out = new StringBuilder();
+            while ((line = reader.readLine()) != null) {
+                out.append(line);
+            }
+            JSONObject json = new JSONObject(out.toString());
+            final OSVAdvisory osvAdvisory = parser.parse(json);
+            if (osvAdvisory != null) {
+                updateDatasource(osvAdvisory);
+            }
+            zipEntry = zipIn.getNextEntry();
+            reader = new BufferedReader(new InputStreamReader(zipIn));
+        }
+        reader.close();
+    }
+
+    public void updateDatasource(final OSVAdvisory advisory) {
+
+        try (QueryManager qm = new QueryManager()) {
+
+            LOGGER.debug("Synchronizing Google OSV advisory: " + advisory.getId());
+            // Set to avoid duplicates
+            Set<VulnerableSoftware> vsList = new HashSet<>();
```

Comment:
Deduplication is required in case the vulnerability with some affected components already exist and we get to parse same vulnerability from OSV but OSV one has few additional affected packages. We dont want to duplicate them.
But yes if Set is not helping here, we'll compare the coming new vulnerable software to any existing one based on its purl info.